moped-editor/src/components/GridTable/GridTableFilters.js

```diff
@@ -416,8 +424,26 @@ const GridTableFilters = ({
   useEffect(() => {
     if (Object.keys(filterParameters).length === 0) {
       filterState.setFilterParameters(filterParameters);
+      // Add an empty filter so the user doesn't have to click the 'add filter' button
+      generateEmptyFilter();
     }
-  }, [filterParameters, filterState]);
+  }, [filterParameters, filterState, generateEmptyFilter]);
+
+  /**
+   * This side effect monitors whether the user has added a complete filter
+   */
+  useEffect(() => {
+    Object.keys(filterParameters).forEach((filterKey) => {
+      if (
+        filterParameters[filterKey].value === null ||
+        filterParameters[filterKey].value === ""
+      ) {
+        setFilterComplete(false);
+      } else {
+        setFilterComplete(true);
+      }
+    });
+  });
```

Comment:
@tillyw this looks good! you just need to add the dependency array at the end, `[filterParameters]`. otherwise i believe it just runs on every render.